### PR TITLE
[FIX] Do not ask credentials if not needed

### LINKED ISSUE #52: install unnecessarily asking for permissions when it has nothing to do
````
u3d install 5.6.1f1
Unity 5.6.1f1 is already installed
Ignoring Unity module, it is already installed
Root privileges are required
Password for lacostej:
````

We really need some unit testing around the various scenarios for package installations.

### PR DESCRIPTION
**Pull request by niezbop**
Returns and skip credentials if no actions are going to be taken.

Fixes #52

### CODE REVIEW

File: lib/u3d/commands.rb [hunk 1]
```diff
@@ -110,6 +110,7 @@ def download(args: [], options: {})
 
         unity = check_unity_presence(version: version)
         return unless enforce_setup_coherence(packages, options, unity)
+        return if packages.empty?
```
Comment (lacostej) [inline lib/u3d/commands.rb]: I am not sure this is what we want. Empty packages mean install everything as well. See `Downloader.download_modules`

Comment (lacostej) [inline lib/u3d/commands.rb]: That's why I said in #52 that we should start having some unit tests :)

Comment (niezbop) [inline lib/u3d/commands.rb]: Just looked at download_modules and I'm not sure this is where we want to parse empty packages as download packages. In my opinion this should be done in the `enforce_setup_coherence`, as it is precisely an operation that enforces make sure the request setup is correct. In any case, interpreting no packages as download all packages should be done before the `enforce_setup_coherence`. If not, we lose an important part of its functionnality and we create counter intuitive behaviours (precisely what I encountered yesterday), where when trying to install the exact setup you have, it filters out all your installed packages, and try to download all the packages, including the one you already have installed.

Comment (niezbop) [inline lib/u3d/commands.rb]: In any case, I fully agree that we need tests to cover every possible scenario available so we make sure that it is behaving precisely as we want to.

**Comment by niezbop** [comment]
This is one of the issue being taken into consideration in #79 

Closing the issue.

**Pull request closed without merging**